rfc/text/advanced/ap_10_rpc_progressive_calls.md

```diff
@@ -1,5 +1,230 @@
+
+{align="left"}
+CALL.Options.sticky|bool := true
+
+
+In this case *Dealer* must make a first `INVOCATION` based on `<invocation_policy>` and then route all next `progressive` 
+calls to the same *Callee*.
+
+If binding all ongoing `progressive` calls to the same *Callee* is not required, *Caller* can set `sticky` option to `FALSE`. 
+
+If `CALL.Options.sticky` is not specified it is treated like `TRUE`, so all `progressive`
+calls go to the same *Callee*.
+
+**Progressive Call Cancellation**
+
+If the original *Caller* is no longer available for some reason (has left the realm), then the *Dealer* will send 
+an `INTERRUPT` to the *Callee*. The `INTERRUPT` will have `Options.mode` set to `"killnowait"` to indicate to the 
+client that no response should be sent to the `INTERRUPT`.
+
+```
+[INTERRUPT, INVOCATION.Request|id, Options|dict]
+```
+Options:
+```
+INTERRUPT.Options.mode|string == "killnowait"
+```
+
+Progressive call cancellation, like progressive call result cancellation closes an important safety gap: 
+In cases where progressive calls are used to stream data from *Callers*, and network connectivity is unreliable, 
+*Callers* my often get disconnected in the middle of sending progressive data. This can lead to unneeded memory
+consumption on *Dealer* and *Callee* sides, because of storing runtime data about ongoing calls.
+
+The message flow for progressive results cancellation involves:
+
+{align="left"}
+,------.            ,------.                 ,------.
+|Caller|            |Dealer|                 |Callee|
+`--+---'            `--+---'                 `--+---'
+   |  CALL (progress)  |                        |
+   | ----------------->|                        |
+   |                   | INVOCATION (progress)  |
+   |                   | ---------------------->|
+   |  CALL (progress)  |                        |
+   | ----------------->|                        |
+   |                   | INVOCATION (progress)  |
+   |                   | ---------------------->|
+,--+---.               |                        |
+|Caller|               |                        |
+`------'               |                        |
+ (gone)                |                        |
+                       |       INTERRUPT        |
+                       | ---------------------->|
+                       |                        |
+                    ,--+---.                 ,--+---.
+                    |Dealer|                 |Callee|
+                    `------'                 `------'
+
+
+Note: Any `ERROR` returned by the *Callee*, in response to the `INTERRUPT`, is ignored (same as in regular call 
+canceling when mode="killnowait"). So, it is not necessary for the *Callee* to send an `ERROR` message.
+
+**Ignoring Progressive Call Requests**
+
+A *Callee* that does not support progressive call SHOULD ignore any `INVOCATION.Details.progress` flag.
```

Comment:
IMO: No, it should kill the connection to the non-compliant router. If either the CALLEE or the DEALER did not announce the respective capability, and the other peer does use the feature anyways, this is violating the agreed protocol behavior, and thus the peer is bogus. Any subsequent talking to a bogus implementation is a security risk, and its semantics are undefined.

Reply:
in general, I am thinking like this: a single call involves 2 sessions (for the 2 connection legs CALLER-DEALER and CALLEE-DEALER), and there are 2 categories of problems:

a) a problem on a single leg, like using progress when it wasn't announced by both peers
b) a problem because one leg uses progress while the other doesn't support it

a) should are protocol errors, but b) is more an application error.